### PR TITLE
feat(workspace): copy codespace url command

### PR DESCRIPTION
**Pull request by Harshita-mindfire**
This PR aims to create a new command `Copy Codespace URL` that would copy the github.dev URL to the clipboard.

Docs PR: https://github.com/dendronhq/dendron-site/pull/635

# Dendron Extended PR Checklist

### Basics

- [x] code should follow [Code Conventions](dev.process.code)
- [x] circular dependency check: make sure your code is not introducing new circular dependencies in plugin-core.  See [Avoiding Circular Dependencies](dev.process.code.best-practices).
- [x] sticking to existing conventions instead of creating new ones
  - eg: [if configuration for utilities are already in one module or package, add future utilities there as well](https://github.com/dendronhq/dendron/pull/1960#discussion_r786228021)

### Extended
- General
  - [x] check whether code be simplified
  - [x] check if similar function already exist in the codebase. if so, can it be re-used?
  - [x] check if this change adversely impact performance
- Operations
  - [x] when shipping this change, will it just work or will it introduce additional operational overhead due to complicated interface or known bugs?
- Architecture
  - [x] check if code is introducing changes on a foundational class or interface. if so, call for design review if needed
    - eg: [making changes to DNode](https://github.com/dendronhq/dendron/pull/2158#pullrequestreview-854689586)

### Basics
- [!] if you are adding analytics related changes, make sure the [Telemetry](https://wiki.dendron.so/notes/84df871b-9442-42fd-b4c3-0024e35b5f3c.html) docs are updated

### Extended
- [!] can we track the performance of this change to know if it is _successful_?
  - eg: [see usage for export pod](https://github.com/dendronhq/dendron/pull/2190#pullrequestreview-855715612)

### Basics

- [x] [Write Tests](dev.process.qa.test) 
- [ ] [Confirm existing tests pass](dev.process.qa.test)
- [x] [Confirm manual testing](dev.process.qa.test) 
- [x] Common cases tested
- [ ] 1-2 Edge cases tested
- [ ] If your tests changes an existing snapshot, snapshots have been [updated](dev.process.qa.test)

### Extended
NA

## Docs
- [x] if your change reflects documentation changes, also submit a PR to [dendron-site](https://github.com/dendronhq/dendron-site) and mention the doc PR link in your current PR
- [~] does this change introduce a new or better way of doing things that others need to be aware of? if so, an async should be created and a process added in [Development](dev) or [Packages](pkg)

### Extended
- [~]  is this a developer BREAKING change? if another person cloning from this branch will need to adjust their dependencies or mental model of the architecture, then it is. if this is the case, make sure this is communicated according to [Close Loop](dev.process.close-loop)
  - eg: [breaking dev change due to new dependency](https://github.com/dendronhq/dendron/pull/2188#pullrequestreview-855696330)

### CODE REVIEW

File: packages/plugin-core/src/commands/CopyCodespaceURL.ts [hunk 1]
```diff
@@ -0,0 +1,54 @@
...
+    const editor = vscode.window.activeTextEditor;
+
+    if (vscode.workspace.workspaceFolders && editor) {
+      const folder = vscode.workspace.getWorkspaceFolder(editor.document.uri);
+
+      if (!folder) {
+        return;
+      }
+
+      const root = (await GitUtils.getGitRoot(folder.uri.fsPath)) || "";
```
Comment (jonathanyeung) [inline packages/plugin-core/src/commands/CopyCodespaceURL.ts:41]: if it's not a git repo, can we have some error message to a user that this command requires a github backed repository to work?

File: packages/plugin-core/src/constants.ts [hunk 2]
```diff
@@ -498,6 +498,11 @@ export const DENDRON_COMMANDS: { [key: string]: CommandEntry } = {
     title: `${CMD_PREFIX} Copy To Clipboard`,
     when: "false",
   },
+  COPY_CODESPACE_URL: {
+    key: "dendron.copyCodespaceURL",
+    title: `${CMD_PREFIX} Copy Codespace URL`,
+    when: `${DendronContext.PLUGIN_ACTIVE} && shellExecutionSupported`,
```
Comment (jonathanyeung) [inline packages/plugin-core/src/constants.ts:504]: it'd be nice if we can add some where clause here to only show this command if the user has a vault / workspace setup that allows it.  Something for later.

**Review by jonathanyeung — CHANGES_REQUESTED** [review]

File: packages/plugin-core/src/commands/CopyCodespaceURL.ts (hunk 1 above)
Comment (Harshita-mindfire) [reply packages/plugin-core/src/commands/CopyCodespaceURL.ts:41]: If not a git repository, it is handled by [getGitRoot](https://github.com/dendronhq/dendron/blob/feat/copy-codespace-url-command/packages/common-server/src/git.ts#L323:L323) util method:

**Review by jonathanyeung — APPROVED** [review]
lgtm apart from merge conflict that crept in

**Review by kevinslin — APPROVED** [review]

**Pull request merged**